packages/semantic-tokens/scripts/util.ts

```diff
@@ -0,0 +1,49 @@
+import { Token } from './token.types';
+
+export const chopLastCamelCasePart = (str: string) =>
+  str
+    .replace(/([a-z])([A-Z])/g, '$1 $2')
+    .split(' ')
+    .slice(0, -1)
+    .join('');
+
+export function removeLastDelimiter(str: string, delimiter: string) {
+  const lastIndex = str.lastIndexOf(delimiter);
+  if (lastIndex === -1) {
+    // Delimiter not found
+    return str;
+  }
+  return str.substring(0, lastIndex);
+}
+
+export function dedupeShadowTokens(_tokenJSON: Record<string, Token>) {
+  /* Our shadow tokens come exported from Figma in parts i.e. X, Y, Blur, Color
+    This is not compatible with our token fallback structure, as V9 shadows are combined strings
```

Comment:
We can probably update this comment based on the test I did: https://codepen.io/brandonthomas/pen/xbxNpYp

Turns out if you have invalid vars it'll toss the initial value and go to the fallback. This means if we didn't have the various parts (x-blur, offsets, etc) in the semantic pieces it would just fallback into the F2 side. While we now know that we can do this, I think it's still fine to keep the existing structure and keep our API surface as small as necessary. If at some point we get feedback we need these we can always add them.